### PR TITLE
Various fixes/improvements for v0.10: Preparation for new scheduling API

### PR DESCRIPTION
**Pull request by skuenzer**
This PR contains various fixes and improvements to the Unikraft core repository for the v0.10 release. To some degree these commits do also prepare the codebase for the new scheduling API. The API will be released separately on a preview branch together with `lib/musl`. However, the changes in this PR are not only `musl` specific and apply also to `newlib` and `pthread-embedded` builds so that we could integrate them into main `staging`.

### CODE REVIEW

File: include/uk/essentials.h [hunk 1]
```diff
@@ -312,6 +320,27 @@ extern "C" {
 #endif
 #endif /* !__containerof */
 
+/* Computes the offsset and size of an field within a struct and checks
```
Comment (razvand) [inline include/uk/essentials.h]:
```suggestion
/* Computes the offsset and size of a field within a struct and checks
```

**Review by razvand — COMMENTED** [review]
Hi, @skuenzer . This looks great. I tested it and all is well. Apart from a minor typo all is well.

@gabrielmocanu please take a look. If it's OK, approve it, add the `...-by` message and then I'll finalize it.

File: plat/xen/arm/link32.lds.S [hunk 2]
```diff
@@ -87,26 +87,7 @@ SECTIONS
...
-	 * just before .bss
-	 */
-
-	.bss : {
-		__bss_start = .;
-		*(.bss)
-		*(.bss.*)
-	}
-	. = ALIGN(__PAGE_SIZE);
-	_end = . ;
```
Comment (gabrielmocanu) [inline plat/xen/arm/link32.lds.S:109]: Does this line need to be deleted?
From what I saw from the link64.lds.S files, this line remained.

File: plat/xen/x86/link64.lds.S [hunk 3]
```diff
@@ -78,25 +78,8 @@ SECTIONS
...
-	 */
+	DATA_SECTIONS
 
-	.bss : {
-		__bss_start = .;
-		*(.bss)
-		*(.bss.*)
-	}
-	. = ALIGN(__PAGE_SIZE);
 	_end = . ;
```
Comment (gabrielmocanu) [inline plat/xen/x86/link64.lds.S]: For consistency.

```suggestion
	_end = .;
```

**Review by gabrielmocanu — CHANGES_REQUESTED** [review]
I tested and everything seems good to me.
I just have a small suggestion related to `_end = .;` for consistency and also I saw that in the `link32.lds.S` file that line was removed, is it how it should be?

File: include/uk/essentials.h (hunk 1 above)
Comment (skuenzer) [reply include/uk/essentials.h]: :+1:

File: plat/xen/x86/link64.lds.S (hunk 3 above)
Comment (skuenzer) [reply plat/xen/x86/link64.lds.S]: :+1: That's fine, I tried to touch as few lines as possible but I agree, it is worth fixing this.

File: plat/xen/arm/link32.lds.S (hunk 2 above)
Comment (skuenzer) [reply plat/xen/arm/link32.lds.S:109]: Uuups, good that you saw this,  it should actually stay.

**Comment by skuenzer** [comment]
Hey, thank you guys that you found those issues. :+1: I pushed an update.

**Review by gabrielmocanu — APPROVED** [review]
All good.

Reviewed-by: Gabi Mocanu <gabi.mocanu98@gmail.com>

**Review by razvand — APPROVED** [review]
Thanks, @skuenzer and @gabrielmocanu 

Reviewed-by: Razvan Deaconescu <razvan.deaconescu@cs.pub.ro>
Approved-by: Razvan Deaconescu <razvan.deaconescu@cs.pub.ro>

**Pull request closed without merging**